internal/impl/redpanda/processor_data_transforms.go

```diff
@@ -0,0 +1,414 @@
+func init() {
+	err := service.RegisterBatchProcessor(
+		"redpanda_data_transform", dataTransformProcessorConfig(),
+		func(conf *service.ParsedConfig, mgr *service.Resources) (service.BatchProcessor, error) {
+			return newDataTransformProcessorFromConfig(conf, mgr)
+		})
+	if err != nil {
+		panic(err)
+	}
+}
+
+//------------------------------------------------------------------------------
+
+type dataTransformConfig struct {
+	inputKey       *service.InterpolatedString
+	outputKeyField *string
+	timestamp      *service.InterpolatedString
+	inputMetadata  *service.MetadataFilter
+	outputMetadata *service.MetadataFilter
+
+	timeout        time.Duration
+	maxMemoryPages int
+}
+
+//------------------------------------------------------------------------------
+
+type dataTransformEnginePool struct {
+	log           *service.Logger
+	wasmBinary    wazero.CompiledModule
+	runtimeConfig wazero.RuntimeConfig
+	modulePool    sync.Pool
+	cfg           dataTransformConfig
+}
+
+func newDataTransformProcessorFromConfig(conf *service.ParsedConfig, mgr *service.Resources) (*dataTransformEnginePool, error) {
+	pathStr, err := conf.FieldString("module")
+	if err != nil {
+		return nil, err
+	}
+
+	fileBytes, err := os.ReadFile(pathStr)
+	if err != nil {
+		return nil, err
+	}
+
+	var cfg dataTransformConfig
+
+	if conf.Contains("input_key") {
+		inputKey, err := conf.FieldInterpolatedString("input_key")
+		if err != nil {
+			return nil, err
+		}
+		cfg.inputKey = inputKey
+	}
+
+	if conf.Contains("output_key") {
+		inputKey, err := conf.FieldString("output_key")
+		if err != nil {
+			return nil, err
+		}
+		cfg.outputKeyField = &inputKey
+	}
+
+	if conf.Contains("input_headers") {
+		inputMetadata, err := conf.FieldMetadataFilter("input_headers")
+		if err != nil {
+			return nil, err
+		}
+		cfg.inputMetadata = inputMetadata
+	}
+
+	if conf.Contains("output_metadata") {
+		outputMetadata, err := conf.FieldMetadataFilter("output_metadata")
+		if err != nil {
+			return nil, err
+		}
+		cfg.outputMetadata = outputMetadata
+	}
+
+	if conf.Contains("timestamp") {
+		ts, err := conf.FieldInterpolatedString("timestamp")
+		if err != nil {
+			return nil, err
+		}
+		cfg.timestamp = ts
+	}
+
+	timeout, err := conf.FieldDuration("timeout")
+	if err != nil {
+		return nil, err
+	}
+	cfg.timeout = timeout
+
+	maxMemoryPages, err := conf.FieldInt("max_memory_pages")
+	if err != nil {
+		return nil, err
+	}
+	cfg.maxMemoryPages = maxMemoryPages
+
+	return newDataTransformProcessor(fileBytes, cfg, mgr)
+}
+
+func newDataTransformProcessor(wasmBinary []byte, cfg dataTransformConfig, mgr *service.Resources) (*dataTransformEnginePool, error) {
+	ctx := context.Background()
+	runtimeCfg := wazero.NewRuntimeConfig().
+		WithCloseOnContextDone(true).
+		WithCompilationCache(wazero.NewCompilationCache()).
+		WithMemoryLimitPages(uint32(cfg.maxMemoryPages))
+	r := wazero.NewRuntimeWithConfig(ctx, runtimeCfg)
+	cm, err := r.CompileModule(ctx, wasmBinary)
+	_ = r.Close(ctx)
```

Comment:
Why swallow the error here instead of returning it if it's non-nil? If this is the right thing to do, I'd consider adding a comment to explain why.

Reply:
Ended up handling the error properly